### ISSUE TITLE
Proxy check events stop updating / events list is stale when handler is added

### ISSUE THREAD

**Issue by mercul3s** [sensu/sensu-go#1250]
Observed issues with event updates on in two different installations and two different networks.

"sensuctl event list" stops showing recent updates. The events displayed stop refreshing and only stale events are visible

"sensuctl entity list" shows recent events. The entitly list shows up-to-date keepalives

The checks continue to run as observed in the sensu logs and via packet traces (tcpdump)

The environment and host info is from @bukiadeniji's sensu install, but I'm also seeing it on an ubuntu 16.04 lxc container I've had running for a week or so. 

## Expected Behavior
Check events are created and routed to the event handler; if the event handler has an error, the check events are still created.

## Current Behavior
Check requests are sent and logged, but check events are not created.

## Your Environment

```
$ sensuctl version
sensu-ctl version 2.0.0-nightly#150c23b, build 150c23b72eb7c2be78773294a2da3397bb781cf7, built 2018-03-09T01:45:03+0000
```
```
$ sensu-backend version
sensu-backend version 2.0.0-nightly#150c23b, build 150c23b72eb7c2be78773294a2da3397bb781cf7, built 2018-03-09T01:43:45+0000
```
```
$ sensu-agent version
sensu-agent version 2.0.0-nightly#150c23b, build 150c23b72eb7c2be78773294a2da3397bb781cf7, built 2018-03-09T01:44:47+0000
```
## Env 1: host5.mydomain

```
[user@host5 log]$ date
Sun Mar 25 13:45:21 PDT 2018
#
#
```
```
[user@host5 ~]$ sensuctl event list
...
...
...
  host1.mydomain.com     check_ssh_ipv4               SSH OK - OpenSSH_7.4 (protocol 2.0) |          0   false      2018-03-23 13:19:37 -0700 PDT
                                                            time=0.010614s;;;0.000000;10.000000
  host1.mydomain.com     check_swap_linux             SWAP OK - Swap usage: 14.500000%,              0   false      2018-03-23 13:18:50 -0700 PDT
                                                            238/1639
  host1.mydomain.com     check_uptime_linux           CheckUptime OK: System has been up             0   false      2018-03-23 13:19:39 -0700 PDT
                                                            since 2018-02-12 21:11:06 +0000
  host1.mydomain.com     keepalive                                                                   0   false      2018-03-23 13:19:33 -0700 PDT
```
```
[user@host5 ~]$ sensuctl entity list
              ID               Class    OS                         Subscriptions                                  Last Seen
 ──────────────────────────── ─────── ─────── ──────────────────────────────────────────────────────── ───────────────────────────────
  host2.mydomain.com   agent   linux   ubuntu,linux,entity:host2.mydomain.com           2018-03-25 13:48:17 -0700 PDT
  host3.mydomain.com      agent   linux   linux,entity:host3.mydomain.com                     2018-03-25 13:48:24 -0700 PDT
  host5.mydomain.com     agent   linux   linux,sensu-server,MON,entity:host5.mydomain.com   2018-03-25 13:48:29 -0700 PDT
  host4.mydomain.com    agent   linux   linux,telem,entity:host4.mydomain.com             2018-03-25 13:48:14 -0700 PDT
  host1.mydomain.com     agent   linux   linux,util,entity:host1.mydomain.com               2018-03-25 13:48:13 -0700 PDT
```
```
[user@host5 ~]$ ps -ef | grep sensu
sensu    24307     1  0 Mar23 ?        00:03:19 /usr/bin/sensu-agent start
sensu    29557     1  0 Mar23 ?        00:16:59 /usr/bin/sensu-backend start -c /etc/sensu/backend.yml
[user@host5 ~]$
```
```
[user@host5 ~]$ sensuctl event info host4.mydomain.com check_disk_usage_linux
=== host4.mydomain.com - check_disk_usage_linux
Entity:    host4.mydomain.com
Check:     check_disk_usage_linux
Output:    DISK OK - free space: / 15327 MB (88.12% inode=99%); /boot 825 MB (81.39% inode=100%);| /=2066MB;17319;17304;0;17394 /boot=188MB;939;924;0;1014
Status:    0
History:   0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
Silenced:  false
Timestamp: 2018-03-23 13:18:39 -0700 PDT
```

################
#################
```
[user@host5 ~]$ sensuctl event list
...
...
...
  host1.mydomain.com     check_memory_linux           MEM OK - system memory usage: 47%              0   false      2018-03-25 14:00:31 -0700 PDT
  host1.mydomain.com     check_ntp_linux              CheckNTP OK: NTP offset by 0.059ms             0   false      2018-03-25 14:00:40 -0700 PDT
  host1.mydomain.com     check_process_linux          CheckProcess OK: Found 102 matching            0   false      2018-03-25 14:00:13 -0700 PDT
                                                            processes
  host1.mydomain.com     check_ssh_ipv4               SSH OK - OpenSSH_7.4 (protocol 2.0) |          0   false      2018-03-25 14:00:39 -0700 PDT
                                                            time=0.012289s;;;0.000000;10.000000
  host1.mydomain.com     check_swap_linux             Last check execution was 114 seconds           1   false      2018-03-25 14:02:44 -0700 PDT
                                                            ago
  host1.mydomain.com     check_uptime_linux           CheckUptime OK: System has been up             0   false      2018-03-25 14:00:40 -0700 PDT
                                                            since 2018-02-12 21:11:06 +0000
  host1.mydomain.com     keepalive                                                                   0   false      2018-03-25 14:00:33 -0700 PDT
```
```
[user@host5 ~]$ ps -ef | grep sensu
sensu    19137     1  7 13:58 ?        00:00:27 /usr/bin/sensu-backend start -c /etc/sensu/backend.yml
sensu    24307     1  0 Mar23 ?        00:03:21 /usr/bin/sensu-agent start
[user@host5 log]$ date
Sun Mar 25 14:04:21 PDT 2018
```

##
##
```
[user@host5 ~]$ sensuctl event list
...
...
...
  host1.mydomain.com     check_ping_proxy             PING OK - Packet loss = 0%, RTA = 0.27               0   false      2018-03-25 15:45:33 -0700 PDT
                                                            ms|rta=0.266000ms;5.000000;10.000000;0.000000
                                                            pl=0%;2;5;0
  host1.mydomain.com     check_process_linux          CheckProcess OK: Found 103 matching processes        0   false      2018-03-25 15:50:13 -0700 PDT

  host1.mydomain.com     check_ssh_ipv4               SSH OK - OpenSSH_7.4 (protocol 2.0) |                0   false      2018-03-25 15:50:37 -0700 PDT
                                                            time=0.008602s;;;0.000000;10.000000
  host1.mydomain.com     check_swap_linux             SWAP OK - Swap usage: 14.400000%, 237/1639           0   false      2018-03-25 15:50:50 -0700 PDT
  host1.mydomain.com     check_uptime_linux           CheckUptime OK: System has been up since             0   false      2018-03-25 15:50:39 -0700 PDT
                                                            2018-02-12 13:11:06 -0800
  host1.mydomain.com     keepalive                                                                         0   false      2018-03-25 15:50:33 -0700 PDT
```
```
[user@host5 ~]$ ps -ef | grep sensu
sensu    24307     1  0 Mar23 ?        00:04:58 /usr/bin/sensu-agent start
sensu    26998     1  0 Mar25 ?        00:07:35 /usr/bin/sensu-backend start -c /etc/sensu/backend.yml
```
```
[user@host5 ~]$ date
Mon Mar 26 11:03:23 PDT 2018
```

#######################
#######################
```
[user@server1 ~]$ sensuctl event list
...
...
...
  server2.my2nddomain.com   check_memory_linux           MEM OK - system memory usage: 14%              0   false      2018-03-23 07:34:32 +0000 UTC
  server2.my2nddomain.com   check_ntp_linux              CheckNTP OK: NTP offset by 0.121ms             0   false      2018-03-23 07:34:40 +0000 UTC
  server2.my2nddomain.com   check_process_linux          CheckProcess OK: Found 214 matching            0   false      2018-03-23 07:34:13 +0000 UTC
                                                          processes
  server2.my2nddomain.com   check_ssh_local_ipv4         SSH OK - OpenSSH_7.4 (protocol 2.0) |          0   false      2018-03-23 07:35:05 +0000 UTC
                                                          time=0.016548s;;;0.000000;10.000000
  server2.my2nddomain.com   check_swap_linux             Last check execution was 120 seconds           1   false      2018-03-23 07:36:50 +0000 UTC
                                                          ago
  server2.my2nddomain.com   check_uptime_linux           CheckUptime OK: System has been up             0   false      2018-03-23 07:34:39 +0000 UTC
                                                          since 2018-01-24 09:17:30 +0000
  server2.my2nddomain.com   keepalive                                                                   0   false      2018-03-23 07:35:00 +0000 UTC
```
```
[user@server1 ~]$ sensuctl entity list
             ID              Class    OS                         Subscriptions                                 Last Seen
 ────────────────────────── ─────── ─────── ─────────────────────────────────────────────────────── ───────────────────────────────
  server3.my2nddomain.com    agent   linux   linux,recursor,entity:server3.my2nddomain.com           2018-03-25 20:45:43 +0000 UTC
  server4.my2nddomain.com    agent   linux   linux,recursor,entity:server4.my2nddomain.com           2018-03-25 20:45:44 +0000 UTC
  server5.my2nddomain.com    agent   linux   linux,recursor,entity:server5.my2nddomain.com           2018-03-25 20:45:47 +0000 UTC
  server6.my2nddomain.com    agent   linux   linux,telemetry,entity:server6.my2nddomain.com          2018-03-25 20:45:37 +0000 UTC
  server1.my2nddomain.com    agent   linux   linux,sensu-server,MON,entity:server1.my2nddomain.com   2018-03-25 20:45:37 +0000 UTC
  server2.my2nddomain.com   agent   linux   linux,mail,zimbra,entity:server2.my2nddomain.com       2018-03-25 20:45:40 +0000 UTC
```
```
[user@server1 ~]$ ps -ef | grep sensu
sensu    10930     1  2 Mar21 ?        01:59:14 /usr/bin/sensu-backend start -c /etc/sensu/backend.yml
sensu    10979     1  0 Mar21 ?        00:12:34 /usr/bin/sensu-agent start
sensu    22612 10979  0 20:39 ?        00:00:00 /bin/bash /etc/sensu/plugins/check_ssh_hosts.sh
sensu    24052 22612  0 20:46 ?        00:00:00 /bin/bash /etc/sensu/plugins/check_ssh_hosts.sh
[user@server1 ~]$
```
```
[user@mon1 log]$ date
Sun Mar 25 20:46:30 UTC 2018
```

**Comment by mercul3s** [comment]
I attempted to replicate this by setting up a test server on ubuntu 16.04, running sensu-backend with the agent running in the foreground, and creating a test http check and slack handler.  I was unable to replicate on a fresh install and saw no errors; check requests and output occurred as expected. I think when I did see an issue in previous testing, it may have been due to the agent not running, though the cause of that crash is unknown. Starting the downed agent resulted in checks executing and handlers handling. Note: I did have an incorrectly configured handler and did not get any indication that the handler was failing; we should be logging in that case. I will file a separate issue to look into that.
